### PR TITLE
Backport west build --pristine to 1.14

### LINKED ISSUE #16414: Backport west build --pristine
Support --pristine on older version to allow for smooth cross-version development using the same tool

### PR DESCRIPTION
**Pull request by carlescufi**
Fixes #16414

### CODE REVIEW

File: scripts/west_commands/build.py [hunk 1]
```diff
@@ -52,6 +53,10 @@
   cmake_opt             Extra options to pass to CMake; implies -c
 '''
 
+class AlwaysIfMissing(argparse.Action):
+
+        def __call__(self, parser, namespace, values, option_string=None):
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply scripts/west_commands/build.py]: No, I do not thing it's worth the risk of changing the code more than it's needed

File: doc/guides/west/config.rst [hunk 2]
```diff
@@ -165,14 +165,15 @@ extension commands (found in :file:`scripts/west_commands`).
...
    * - ``build.pristine``
-     - String. Controls the way in which ``west build`` may run the ``pristine``
-       target before building. Can take the following values:
-
-         - ``never`` (default): Never automatically run the ``pristine`` target.
-         - ``auto``:  ``west build`` will automatically run the ``pristine``
-           target before building, if a build system is present and the build
-           will fail otherwise (e.g. the user has specified a different board or
-           application from the one previously used to make the build
+     - String. Controls the way in which ``west build`` may clean the build
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply doc/guides/west/config.rst]: Same

**Review by dbkinder — APPROVED** [review]
doc changes look OK.

File: scripts/west_commands/build.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbolivar) [reply scripts/west_commands/build.py]: I mean, it's whitespace changes, but fine 😉

File: doc/guides/west/config.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbolivar) [reply doc/guides/west/config.rst]: It's documentation, the risk analysis is not the same and I disagree here.

**Review by mbolivar — CHANGES_REQUESTED** [review]
I think we should keep documentation looking the same for backports and mainline as much as possible.

**Comment by carlescufi** [comment]
@mbolivar addressed the additional backport of linting and doc

**Review by mbolivar — APPROVED** [review]
LGTM, thanks!

**Pull request merged**